prefect_gcp/workers/vertex.py

```diff
@@ -0,0 +1,633 @@
+        ContainerSpec,
+        CustomJob,
+        CustomJobSpec,
+        Scheduling,
+        WorkerPoolSpec,
+    )
+    from google.cloud.aiplatform_v1.types.job_state import JobState
+    from google.cloud.aiplatform_v1.types.machine_resources import DiskSpec, MachineSpec
+    from google.protobuf.duration_pb2 import Duration
+    from tenacity import retry, stop_after_attempt, wait_fixed, wait_random
+except ModuleNotFoundError:
+    pass
+
+_DISALLOWED_GCP_LABEL_CHARACTERS = re.compile(r"[^-a-zA-Z0-9_]+")
+
+if TYPE_CHECKING:
+    from prefect.client.schemas import FlowRun
+    from prefect.server.schemas.core import Flow
+    from prefect.server.schemas.responses import DeploymentResponse
+
+
+class VertexAIWorkerVariables(BaseVariables):
+    """
+    Default variables for the Vertex AI worker.
+
+    The schema for this class is used to populate the `variables` section of the default
+    base job template.
+    """
+
+    region: str = Field(
+        description="The region where the Vertex AI Job resides.",
+        example="us-central1",
+    )
+    image: str = Field(
+        title="Image Name",
+        description=(
+            "The URI of a container image in the Container or Artifact Registry, "
+            "used to run your Vertex AI Job. Note that Vertex AI will need access"
+            "to the project and region where the container image is stored. See "
+            "https://cloud.google.com/vertex-ai/docs/training/create-custom-container"
+        ),
+        example="gcr.io/your-project/your-repo:latest",
+    )
+    credentials: Optional[GcpCredentials] = Field(
+        title="GCP Credentials",
+        default_factory=GcpCredentials,
+        description="The GCP Credentials used to initiate the "
+        "Vertex AI Job. If not provided credentials will be "
+        "inferred from the local environment.",
+    )
+    machine_type: str = Field(
+        title="Machine Type",
+        description=(
+            "The machine type to use for the run, which controls "
+            "the available CPU and memory. "
+            "See https://cloud.google.com/vertex-ai/docs/reference/rest/v1/MachineSpec"
+        ),
+        example="n1-standard-4",
+    )
+    accelerator_type: Optional[str] = Field(
+        title="Accelerator Type",
+        description=(
+            "The type of accelerator to attach to the machine. "
+            "See https://cloud.google.com/vertex-ai/docs/reference/rest/v1/MachineSpec"
+        ),
+        example="NVIDIA_TESLA_K80",
+    )
+    accelerator_count: Optional[int] = Field(
+        title="Accelerator Count",
+        description=(
+            "The number of accelerators to attach to the machine. "
+            "See https://cloud.google.com/vertex-ai/docs/reference/rest/v1/MachineSpec"
+        ),
+    )
+    boot_disk_type: str = Field(
+        title="Boot Disk Type",
+        description="The type of boot disk to attach to the machine.",
+        example="pd-ssd",
```

Comment:
Can this be a default instead of an example?
```suggestion
        default="pd-ssd",
```